### PR TITLE
[Enhancement] information_schema table_configs add two columns, table_engine table_model

### LINKED ISSUE #11659: information_schema.table_configs add columns

### PR DESCRIPTION
**Pull request by waittttting**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #11659

information_schema table_configs add two columns, table_engine table_model

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by waittttting** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 6 / 6 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/service/InformationSchemaDataSource.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/SchemaTable.java|2|2|100.00%|[]|

**Review by Astralidea — DISMISSED** [review]

File: gensrc/thrift/FrontendService.thrift [hunk 1]
```diff
@@ -992,13 +992,15 @@ struct TGetTablesConfigResponse {
 struct TTableConfigInfo {
     1: optional string table_schema
     2: optional string table_name
-    3: optional string primary_key
-    4: optional string partition_key
-    5: optional string distribute_key
-    6: optional string distribute_type
-    7: optional i32 distribute_bucket
-    8: optional string sort_key
-    9: optional string properties
```
Comment (gengjun-git) [inline gensrc/thrift/FrontendService.thrift:1001]: newly added properties should use new id.

**Review by gengjun-git — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: gensrc/thrift/FrontendService.thrift [hunk 2]
```diff
@@ -992,13 +992,15 @@ struct TGetTablesConfigResponse {
...
+    3: optional string table_engine
```
Comment (chaoyli) [inline gensrc/thrift/FrontendService.thrift:995]: If change the id, how to make it compatibility when upgrading?

File: be/src/exec/vectorized/schema_scanner/schema_tables_config_scanner.cpp [hunk 3]
```diff
@@ -99,67 +101,87 @@ Status SchemaTablesConfigScanner::fill_chunk(ChunkPtr* chunk) {
             break;
         }
         case 3: {
-            // primary_key
+            // table_engine
```
Comment (nshangyiming) [inline be/src/exec/vectorized/schema_scanner/schema_tables_config_scanner.cpp:104]: Shall we simplify the `fill_chunk()` function like this, currently there is too many redundant codes.

```c++
    constexpr int MAX_SLOT_IDX = 11;
    for (const auto& [slot_id, index] : slot_id_to_index_map) {
        ColumnPtr column = slot_id <= MAX_SLOT_IDX ? (*chunk)->get_column_by_slot_id(slot_id) : nullptr;
        const std::string* str = nullptr;
        switch (slot_id) {
        case 1: {
            // table_schema
            const std::string* str = &info.table_schema;
            break;
        }
        case 2: {
            // table_name
            const std::string* str = &info.table_name;
            break;
        }
        ...
        case MAX_SLOT_IDX:
        case default:
            break;
        }
        if (column != nullptr) {
            Slice value(str->c_str(), str->length());
            fill_column_with_slot<TYPE_VARCHAR>(column.get(), (void*)&value);
        }
    } // end for loop
```

**Review by nshangyiming — APPROVED** [review]

File: gensrc/thrift/FrontendService.thrift (hunk 2 above)
Comment (waittttting) [reply gensrc/thrift/FrontendService.thrift:995]: This func not yet released

**Comment by waittttting** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 25 / 56 (44.64%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/DynamicPartitionProperty.java|0|18|00.00%|[151, 152, 153, 154, 155, 156, 157, 158, 159, 160, 161, 163, 164, 165, 166, 168, 169, 170]|
|:large_blue_circle:|com/starrocks/service/InformationSchemaDataSource.java|23|36|63.89%|[150, 161, 169, 170, 171, 172, 173, 174, 175, 186, 189, 195, 196]|
|:large_blue_circle:|com/starrocks/catalog/SchemaTable.java|2|2|100.00%|[]|

**Comment by waittttting** [comment]
run starrocks_admit_test

**Review by gengjun-git — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Pull request merged**